tests/KeyOfArrayTest.php

```diff
@@ -0,0 +1,189 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Psalm\Tests;
+
+use Psalm\Tests\Traits\InvalidCodeAnalysisTestTrait;
+use Psalm\Tests\Traits\ValidCodeAnalysisTestTrait;
+
+class KeyOfArrayTest extends TestCase
+{
+    use InvalidCodeAnalysisTestTrait;
+    use ValidCodeAnalysisTestTrait;
+
+    /**
+     * @return iterable<string,array{code:string,assertions?:array<string,string>,ignored_issues?:list<string>,php_version?:string}>
+     */
+    public function providerValidCodeParse(): iterable
+    {
+        return [
+            'keyOfListClassConstant' => [
+                'code' => '<?php
+                    class A {
+                        const FOO = [
+                            \'bar\'
```

Comment:
Please use either double quotes for strings literals or double quotes for quoting the entire snippet instead of escaping. This is way easier when copy/pasting a snippet down the line

Reply:
@orklah I've thought about refactoring tests to use nowdocs for code, unfortunately the indented end identifier requires PHP 7.3. Maybe for Psalm 5?

Reply:
I just tried, I'm not really a fan of how PHPStorm handle that. You have to put the `<?php` on a new line and not indented for coloration to work. Then it tells you the end of nowdoc should have the same indentation (so none). That looks more messy at the end

Reply:
I replaced it with double quotes inside.